README.md

```diff
@@ -0,0 +1,101 @@
+# Fastify Secrets HashiCorp
+
+![CI](https://github.com/nearform/fastify-secrets-hashicorp/workflows/CI/badge.svg)
+
+Fastify secrets plugin for [HashiCorp Vault](https://www.vaultproject.io).
+
+## Installation
+
+```sh
+npm install --save fastify-secrets-hashicorp
+```
+
+## Usage
+
+### Add plugin to your fastify instance
+
+```js
+fastify.register(FastifySecretsHashiCorp, {
+  secrets: {
+    dbPassword: 'secret-name'
+  }
+})
+```
+
+### Access your secrets
+
+```js
+await fastify.ready()
+
+console.log(fastify.secrets.dbPassword) // content of 'secret-name'
+```
+
+### Plugin options
+
+The plugin can be initialised with clientOptions as follows:
+
+```js
+fastify.register(FastifySecretsHashiCorp, {
+  secrets: {
+    dbPassword: 'secret-name'
+  },
+  clientOptions: {
+  }
+})
+```
+
+Following the structure
+
+```js
+clientOptions: {
+    vaultOptions: {
+      token: '', 
+      endpoint: ''
+    },
+    mountPoint, ''
+  }
+```
+
+#### clientOptions.mountPoint
+
+The [path to the secrets engine](https://www.vaultproject.io/docs/secrets#secrets-engines-lifecycle). Defaults to 'secret'.
+
+#### clientOptions.vaultOptions
+
+Initialisation options that are sent to [node-vault](https://github.com/kr1sp1n/node-vault), typed as [VaultOptions](https://github.com/kr1sp1n/node-vault/blob/master/index.d.ts#L115).
+
+The most important being:
+
+- vaultOptions.token: Vault access token. Defaults to process.env.VAULT_TOKEN.
+- vaultOptions.endpoint: Endpoint to the Vault API. Defaults to process.env.VAULT_ADDR else 'http://127.0.0.1:8200'
+
+### Assumptions
+
+- The vault is unsealed
+- The secrets engine is [kv-v1](https://www.vaultproject.io/docs/secrets/kv/kv-v1)
```

Comment:
https://github.com/mercurius-js/mercurius/releases/tag/v7.1.0

Reply:
Did you paste the right thing here?

Reply:
probably not :)